### ISSUE TITLE
[Feature request]: Provide a way to write note guid into exported .enex

### ISSUE THREAD

**Issue by coldwinds** [vzhd1701/evernote-backup#66]
### Suggestion

Note guid is an important information to identify a note, and good help to rebuild link relationship when a note have a link to another note, with guid in exported .enex for every note make it possible to rebuild the connection when import into 3rd party apps.

**Comment by benjiwheeler** [comment]
Is there a place within a given .enex file that you think it would make sense for a guid to go?

**Comment by AltoRetrato** [comment]
One issue with this approach is that adding a new attribute to a .enex file, which is an XML file structured according to its [DTD](http://xml.evernote.com/pub/evernote-export4.dtd), could potentially make the file invalid in many programs. Some programs may not enforce strict DTD validation, allowing the modification. A workaroud could be using a modified DTD, but that may still cause compatibility issues with software that expects strict conformance to the original DTD.

I would also like to retain the GUID, and I thought it could be included in the `<note-attributes>` element, such as within a `<note-id>` element:

```
  <note>
    <title>[...]</title>
    <created>[...]</created>
    <updated>[...]</updated>
    <note-attributes>
      <author>[...]</author>
      <source>[...]</source>
      <source-application>[...]</source-application>
      <note-id>[...]</note-id>
    </note-attributes>
    <content>
      [...]
    </content>
  </note>
```

But I believe such modification should be optional in order to not break compatibility with other programs, specially when you consider that no program currently uses such data. And I think it would be easier to access the data directly in the `en_backup.db` SQLIte DB if you want to use it to export to other file formats.

**Comment by bumper314** [comment]
I agree, exporting the GUID in the enex is a major necessity and would really simplify the work of conversion tools like https://github.com/akosbalasko/yarle/.

**Comment by kranix0** [comment]
Thanks @vzhd1701 for getting this done. Does this mean that third-party tools like [Yarle](https://github.com/akosbalasko/yarle) will have what they need to maintain links between notes? Are there any circumstances where this will not be possible?

**Comment by vzhd1701** [comment]
`--add-guid` adds a non-standard `guid` tag into each note in ENEX file. I might be wrong, but as I understood from a quick glance at [yarle](https://github.com/akosbalasko/yarle) code, it is using notes names and a special "Table of Contents" note to backtrack the notes GUIDs. I think @akosbalasko will have to implement a special case which will try to lookup guid tag directly from the note tags in ENEX file. Or I can add another flag to generate that "Table of Contents" note with the links to each note in the notebook.

**Comment by AltoRetrato** [comment]
I already wrote a conversion tool using the evernote-backup database instead of .EXEX files, so it can keep all links among all notes, as long as you export all linked notes at the same time.
